### PR TITLE
Removing dependency on simple-oauth2 to simplify example.

### PR DESCRIPTION
**Pull request by neverendingqs**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -43,39 +35,44 @@ app.get('/', function(req, res) {
...
 app.post('/auth', function(req, res) {
-    var authorization_uri = oauth2.authCode.authorizeURL({
-        redirect_uri: getRedirectUri(req),
-        scope: req.body.scope,
-        // Generate a secure state in production to prevent CSRF (https://tools.ietf.org/html/rfc6749#section-10.12)
-        state: "f4c269a0-4a69-43c1-9405-86209c896fa0"
-    });
-
-    res.redirect(authorization_uri);
+    var authCodeParams = encodeURI(
```
Comment (AlexBedley) [inline server.js]: Would using the built-in querystring module clear this up? https://nodejs.org/api/querystring.html#querystring_querystring_stringify_obj_sep_eq_options

File: server.js [hunk 2]
```diff
@@ -43,39 +35,44 @@ app.get('/', function(req, res) {
...
+        .end(function(err, postResponse) {
+            if (err) {
+                console.log('Access Token Error', error.message);
+                res.redirect('/');
+            } else {
+                // We are storing the access token in a cookie for simplicity, but the user agent should never have to see it
+                res.cookie(cookieName, { accessToken: postResponse.body.access_token }, cookieOptions);
+
+                // Optionally, store the refresh token (postResponse.body.refresh_token) to a user context (https://tools.ietf.org/html/rfc6749#section-6)
+
```
Comment (AlexBedley) [inline server.js]: extra whitespace?

**Comment by AlexBedley** [comment]
:shipit:

File: server.js (hunk 2 above)
Comment (neverendingqs) [inline server.js]: Intended, because it's a separate thing that could be done if desired.

File: server.js (hunk 1 above)
Comment (neverendingqs) [inline server.js]: I was a bit torn (leaving it like this makes it easier to read for non-JS devs, but I went with the change (should be easy to look up the method).

**Comment by AlexBedley** [comment]
:shipit:

**Pull request merged**